drafts/building-github-pull-requests-with-jenkins-pipelines.md

```diff
@@ -0,0 +1,114 @@
+
+To get started navigate to the `New Item` page in Jenkins and configure as follows:
+
+![](https://kabisa-website-assets.s3.amazonaws.com/theguild/posts/building-github-pull-requests-with-jenkins-pipeline-style/create-organisation-directory.png)
+
+On the next screen some details about your GitHub origanisation have to be configured:
+
+![](https://kabisa-website-assets.s3.amazonaws.com/theguild/posts/building-github-pull-requests-with-jenkins-pipeline-style/configure-github-organisation.png)
+
+* **Owner**: enter the name of your GitHub organisation here
+* **Scan credentials**: select or create credentials that should be used to communicate with the GitHub API. You should use a GitHub OAuth token here that has permissions to access all repositories you'd like to build in Jenkins.
+* **Checkout credentials**: Configures which credentials should be used to clone repositories. By default https will be used using the same OAuth token used for scanning but you can also chose to use an SSH key here.
+
+Adapt the other fields to your own taste. With the configuration shown in the screenshot Jenkins will build only the `master` branch of each project and all Pull Requests.
+
+Hit the save button and Jenkins will start scanning your GitHub account for repositories containing a `Jenkinsfile` in the project root. `Jenkinsfile`s tell Jenkins how your project should be built. If this is your first interaction with Jenkins Pipelines you probably won't have any projects that have a Jenkinsfile so read on to learn how to create a Jenkinsfile.
+
+Jenkins is now fully setup to build any project in your GitHub organisation, provided it has a `Jenkinsfile`. Contrary to Jenkins 1.x no further interaction with Jenkins web UI is required to enable and configure new projects. Simply add `Jenkinsfile`s to your projects and Jenkins will automatically start building your project! 
+
+## Creating a Jenkinsfile
+
+A `Jenkinsfile` tells Jenkins how to build your project. Jenkinsfiles are very powerful as you'll have the full power of [Groovy](http://www.groovy-lang.org/) at your fingertips. I won't go into detail about the Jenkinsfile syntax as that's beyond the scope of this post. [Here's](https://jenkins.io/doc/pipeline/) some documentation to get you started. Read on for a simple example.
+
+The most basic `Jenkinsfile` looks like this:
+
+```groovy
+node {
+    echo 'Hello from Pipeline'
+}
+```
+
+Obviously this doesn't do much useful, but bear with me. Put this in a file named `Jenkinsfile` in the root of a Github project, commit and push. With this your project can now be built by a Jenkins pipeline, however Jenkins doesn't know about this project yet. By default Jenkins will only look for new projects once a day. To ensure the project is recognized immediately you can press the `Re-scan Organization > Run Now` button. Jenkins will then scan the GitHub organisation again and this time find the `Jenkinsfile` we just commited and setup the project for you.
```

Comment:
... Jenkins will then scan the GitHub organisation again, this time finding the Jenkinsfile we've just committed and setting up the project for you.